source/RValue.h

```diff
@@ -0,0 +1,176 @@
+/* RValue.h
+Endless Sky is free software: you can redistribute it and/or modify it under the
+terms of the GNU General Public License as published by the Free Software
+Foundation, either version 3 of the License, or (at your option) any later version.
+
+Endless Sky is distributed in the hope that it will be useful, but WITHOUT ANY
+WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+PARTICULAR PURPOSE. See the GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License along with
+this program. If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#ifndef RVALUE_H_
+#define RVALUE_H_
+
+
+#include <stdexcept>
+#include <string>
+
+
+#define DEBUG_RVALUE_CONDITIONS
+
+
+
+// This class stores an rvalue and remembers where the rvalue came
```

Comment:
It seems odd to give and int or a string special meaning depending on its source. Seems like that is just a way to smuggle context, and then by doing so couple that context to what ever method uses the r value key to do conditional logic.

Reply:
Yes. You need the context so you can reapply the condition variable after it updates.